tasks/get_sql_logins.ps1

```diff
@@ -10,110 +10,25 @@ param (
   [switch]$detailed
 )
 
-$error = @{
-  _error = @{
-    msg     = ''
-    kind    = 'puppetlabs.task/task-error'
-    details = @{
-      detailedInfo = ''
-      exitcode     = 1
-    }
-  }
-}
-
-function Select-LoginName {
-  param(
-    [PSObject]$login,
-    [string[]]$namesToMatch,
-    [switch]$exact_match
-  )
-
-
-  # This function takes a single SQLServer login object and compares it against
-  # the list of names passed into the -login_name parameter of the script to
-  # determine if this is a login the user is interested in seeing. If it does
-  # not pass the filter represented by that parameter the login is discarded.
-
-  foreach ($paramLogin in $namesToMatch) {
-    if ($exact_match) {
-      if ($paramLogin -eq $login.name) {
-        Write-Output $login
-      }
-    }
-    else {
-      # Match is a regex operator, and it doesn't like the '\' in domain names.
-      if ($login.name -match [regex]::escape($paramLogin)) {
-        Write-Output $login
-      }
-    }
-  }
-}
-
-function Get-SQLInstances {
-  param(
-    [string[]]$instance_name
-  )
-
-  $instancesHolder = New-Object System.Collections.Generic.List[System.Object]
-  $stringsToReturn = New-Object System.Collections.Generic.List[System.Object]
-
-  # The default instance is referred to in its service name as MSSQLSERVER. This
-  # leads many SQLSERVER people to refer to it as such. They will also connect
-  # to it using just a '.'. None of these are its real name. Its real instance
-  # name is just the machine name. A named instances real name is the machine
-  # name a, '\', and the instance name. This little foreach ensures that we are
-  # referring to these instances by their real names so that proper filtering
-  # can be done.
-
-  foreach ($name in $instance_name) {
-    switch ($name) {
-      {($_ -eq 'MSSQLSERVER') -or ($_ -eq '.')} { [void]$instancesHolder.add($env:COMPUTERNAME) }
-      {$_ -eq $env:COMPUTERNAME} { [void]$instancesHolder.add($_) }
-      {$_ -notmatch '\\'} { [void]$instancesHolder.add("$env:COMPUTERNAME\$_") }
-      default { [void]$instancesHolder.add($name) }
-    }
-  }
-
-  $instanceStrings = (Get-ItemProperty 'HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server').InstalledInstances
-
-  # The registry key does not return the real instance names. Again we must
-  # normalize these names into their real names so that comparisons can be done
-  # properly.
-
-  foreach ($string in $instanceStrings) {
-    switch ($string) {
-      'MSSQLSERVER' { $string = $env:COMPUTERNAME }
-      Default {$string = "$env:COMPUTERNAME\$string"}
-    }
-
-    if ((-not [string]::IsNullOrEmpty($instancesHolder))-and(-not [string]::IsNullOrWhiteSpace($instancesHolder))) {
-      foreach ($instance in $instancesHolder) {
-        if ($instance -eq $string) {
-          [void]$stringsToReturn.add($string)
-        }
+$currentFolder = Split-Path -parent $PSCommandPath
+
+$helpersFilePath = "$currentFolder\..\files\shared_task_functions.ps1"
```

Comment:
Since we're using `..` notation here, should we split $CurrentFolder again to get the parent folder?

```powershell
$currentFolder = Split-Path -parent $PSCommandPath
$parentFolder  = Split-Path -parent $currentFolder
```

Reply:
Yeah I don't mind doing this.